docs/proposals/acting-as.md

```diff
@@ -0,0 +1,219 @@
+the accounts it is allowed to act as, and when a namespace is created,
+either the namespace admin consents to let controller-manager act in it,
+or a namespace initializer process creates the account.
+
+This is a good approach too.
+It is analgous to creating a "bot" account on our github organization, and then adding that
+bot account to the ACLs of our github repo.
+
+Good things about this approach:
+
+1. no special purpose header needed
+1. no need to record two users in an audit log (original and acting-as).
+1. no need for second authorization stage in the apiserver.
+1. since the special-purpose account (e.g. "controller-manager-acting-in-namespace-foo")
+   is added to the ACL for a particular namespace, a Policy Manager for that namespace
+   could revoke that ACL if desired.
+
+Concerns about this approach:
+
+1. more complex to get right at first.
+1. requires that users see a special purpose account added to the ACL of their namespace/cluster
+   for each use-case.  
+1. workflow for automating adding these to new namespaces could get complex.  Even more
+   so when there are multiple roles in a namespace.
+1. automation of the workflow for adding these special-purpose accounts to ACLs requires
+   a standardized ACL API for kubernetes, which competes with the goal of pluggable
+   authorization in kubernetes.
+1. figuring out where these special accounts fit into existing accounts and auditing frameworks.
+
+Recommendation: keep in mind as something we may later evolve to.
+
+### Alternative: Oauth-like
+
+With Oauth, a user delegates his own authority to access his personal data.
+In a kubernetes cluster, multiple users have access to a namespace,
+and multiple users may modify a pod.  And, in the future, those users
+may have different permissions within the namespace.  And one of those
+users might leave the company, but the replication controller needs
+to keep working.  So, traditional oauth flows do not seem like a
+good fit.  Experience of my colleagues is that ACLs are better.
+Products like Amazon IAM notably do not use oauth as a core part
+of their solution.
+
+Recommendation: reject this alternative.
+
+### Alternative: Acting-as the human who create the RC
+
+In this alternative, when the RC is created, the creating human user
+is recorded in either the metadata of the RC or in the spec
+of the RC.  When the RC-manager creates a pod, it acts as
+that human user (or some intersection of the human and other
+things).
+
+Consider what happens if a human user to creates a replication controller
+for an important service, and then leave the company.
+The employee no longer appears in the user database, and so the apiserver
+can no longer recognize that human user, and so it will refuse to
+allow the RC-manager to act as that user.   Or the user still appears in the user
+database, but was removed from all ACLs on termination.  So, again
+the RC-manager cannot create pods.
+
+Recommendatin: reject this alternative.
```

Comment:
I agree, reject.  It's also unclear in this sort of an approach, what should happen if someone else edits the resource.